File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int answer = 0;
vector<int> v;

void dfs(int cnt, int sum, int t){
    if(cnt == v.size()){
        if(sum == t){
            answer++;
        }
        return;
    }
    dfs(cnt + 1, sum + v[cnt], t);
    dfs(cnt + 1, sum - v[cnt], t);
}

int solution(vector<int> numbers, int target) {
    v = numbers;
    dfs(0, 0, target);
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let answer = 0;
    
    const dfs = (cnt, sum) => {
        if(cnt === numbers.length){
            if(sum === target){
                answer++;
            }
            return;
        }
        
        dfs(cnt + 1, sum + numbers[cnt]);
        dfs(cnt + 1, sum - numbers[cnt]);
    }
    
    dfs(0, 0);
    return answer;
}